api/v1alpha1/clusterconfig_types.go

```diff
@@ -344,6 +348,66 @@ func (ImageRegistries) VariableSchema() clusterv1.VariableSchema {
 	}
 }
 
+type Users []User
+
+func (Users) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Description: "Users to add to the machine",
+			Type:        "array",
+			Items:       ptr.To(User{}.VariableSchema().OpenAPIV3Schema),
```

Comment:
Why ptr.To?

Reply:
Good question. We can't just reference the return value of `VariableSchema()`, although this has been [proposed](https://github.com/golang/go/issues/45653), but apparently turned down.